Gems/Atom/RPI/Code/Source/RPI.Public/Pass/CopyPass.cpp

```diff
@@ -109,14 +181,98 @@ namespace AZ
+                        ->WaitOnCpu();
+                }
+                for (auto& fence : m_device2WaitFence)
+                {
+                    fence
+                        ->GetDeviceFence(
+                            m_data.m_destinationDeviceIndex != RHI::MultiDevice::InvalidDeviceIndex ? m_data.m_destinationDeviceIndex
+                                                                                                    : RHI::MultiDevice::DefaultDeviceIndex)
+                        ->WaitOnCpu();
+                }
+            }
+        }
+
         // --- Scope producer functions ---
 
-        void CopyPass::SetupFrameGraphDependencies(RHI::FrameGraphInterface frameGraph)
+        void CopyPass::SetupFrameGraphDependenciesSameDevice(RHI::FrameGraphInterface frameGraph)
         {
-            RenderPass::SetupFrameGraphDependencies(frameGraph);
+            for (const PassAttachmentBinding& attachmentBinding : m_attachmentBindings)
+            {
+                if (attachmentBinding.GetAttachment() != nullptr &&
+                    frameGraph.GetAttachmentDatabase().IsAttachmentValid(attachmentBinding.GetAttachment()->GetAttachmentId()))
+                {
+                    switch (attachmentBinding.m_unifiedScopeDesc.GetType())
+                    {
+                    case RHI::AttachmentType::Image:
+                        {
+                            frameGraph.UseAttachment(
+                                attachmentBinding.m_unifiedScopeDesc.GetAsImage(),
+                                attachmentBinding.GetAttachmentAccess(),
+                                attachmentBinding.m_scopeAttachmentUsage);
+                            break;
+                        }
+                    case RHI::AttachmentType::Buffer:
+                        {
+                            frameGraph.UseAttachment(
+                                attachmentBinding.m_unifiedScopeDesc.GetAsBuffer(),
+                                attachmentBinding.GetAttachmentAccess(),
+                                attachmentBinding.m_scopeAttachmentUsage);
+                            break;
+                        }
+                    default:
+                        AZ_Assert(false, "Error, trying to bind an attachment that is neither an image nor a buffer!");
+                        break;
+                    }
+                }
+            }
+            for (Pass* pass : m_executeAfterPasses)
+            {
+                RenderPass* renderPass = azrtti_cast<RenderPass*>(pass);
+                if (renderPass)
+                {
+                    frameGraph.ExecuteAfter(renderPass->GetScopeId());
+                }
+            }
+            for (Pass* pass : m_executeBeforePasses)
+            {
+                RenderPass* renderPass = azrtti_cast<RenderPass*>(pass);
+                if (renderPass)
+                {
+                    frameGraph.ExecuteBefore(renderPass->GetScopeId());
+                }
```

Comment:
Can we remove this? Ideally we want to remove ExecuteAfter/ExecuteBefore completely from Atom as no one should be using them (its in the backlog). They imply execution dependency between passes which is not true. The execution dependency on the gpu is setup via resources setup as attachments and not via emitting gpu commands for one pass before another.